frontend/src/components/PurchasedBooks/Purchased.js

```diff
@@ -0,0 +1,35 @@
+
+import React, { useContext } from 'react';
+import BookContext from '../../contexts/BookContext.js';
+import classes from './Purchased.module.css';
+
+export default function Purchased() {
+    const { purchasedBooks } = useContext(BookContext);
+
+    return (
+        <>
+            <div className={classes.container}>
+                <h1 className={classes.siteName}>Design Books</h1>
+
+            </div>
+            <section className={classes.purchSection}>
+                <ul className={classes.purchul}>
+                    {purchasedBooks.length > 0 ? (
+                        purchasedBooks.map((book, index) => {
+                            return (
+                                <li key={index}>
+                                    <button className={classes.purchdata}>
+                                        <img src={book.volumeInfo.imageLinks?.thumbnail} alt={book.volumeInfo.title} className={classes.bookimage} />
+                                        <p>{book.volumeInfo.title}</p>
```

Comment:
Handle cases where book.volumeInfo.imageLinks or book.volumeInfo.title might be undefined
like
<img
                    src={book.volumeInfo?.imageLinks?.thumbnail || 'default-thumbnail.jpg'}
                    alt={book.volumeInfo?.title || 'No Title'}
                    className={classes.bookimage}
                  />